addons/Animation/src/presenter.js

```diff
@@ -60,6 +64,55 @@ function AddonAnimation_create (){
         return upgradedModel;
     };
 
+    presenter.addTextToSpeech = function (model) {
+        var upgradedModel = {};
+        $.extend(true, upgradedModel, model);
+
+        if (!upgradedModel['Alternative Text']) {
+            upgradedModel['Alternative Text'] = ''
+        }
+
+        if (!upgradedModel['Preview Alternative Text']) {
+            upgradedModel['Preview Alternative Text'] = ''
+        }
+
+        if (upgradedModel['speechTexts'] === undefined) {
+            upgradedModel['speechTexts'] = {
+                Stop: {Stop: "stop"}
+            };
+        }
+
+        if (upgradedModel['langAttribute'] === undefined) {
+            upgradedModel['langAttribute'] = '';
+        }
+
+        return upgradedModel;
+    };
+
+    function getSpeechTextProperty (rawValue, defaultValue) {
+        var value = rawValue.trim();
+
+        if (value === undefined || value === null || value === '') {
+            return defaultValue;
+        }
+
+        return value;
+    }
+
+    presenter.setSpeechTexts = function(speechTexts) {
+        presenter.speechTexts = {
```

Comment:
Po co to przypisanie tutaj? Upgrade modelu tego nie załatwił?

Reply:
Upgrade modelu robi tylko to co nazwa wskazuje - sprawdza czy speechTexts jest obecne, a jeśli nie to uzupełnia brakujące pola modelu z domyślnymi wartościami. Niczego nie robi natomiast w sytuacji, gdzie speechTexts jest obecny, ale użytkownik nie ustawił w nim żadnych wartości. Ta definicja 1) definiuje wartości domyślne 2) w sposób łatwy do odczytania i 3) zgodny z konwencją stosowaną dla TTS jeszcze od Maćka.